File: examples/todo/todo.ts
import { Instant } from './todo.extern';

export type UUID = string & {
  type: 'uuid'
};

export class Todo {
  id: UUID;
  title: string;
  createdAt: Instant;
  checkedAt: (Instant | null);
}

export class CreateTodo {
  title: string;
}

File: examples/todo/todo.dart
/* UNSUPPORTED */
/* UNSUPPORTED */
class Todo {
  UUID id;
  String title;
  Instant createdAt;
  Instant? checkedAt;

  Todo({
    required this.id,
    required this.title,
    required this.createdAt,
    this.checkedAt,
  });

  static void $hWrite(Writer writer,value:Todo) => {}
  factory Todo.$hRead(Reader reader) => {}
  static Schema $hSchema = {};
}

class CreateTodo {
  String title;

  CreateTodo({
    required this.title,
  });

  static void $hWrite(Writer writer,value:CreateTodo) => {}
  factory CreateTodo.$hRead(Reader reader) => {}
  static Schema $hSchema = {};
}
